cpp/tests/orderby/is_sorted/is_sorted_tests.cu

```diff
@@ -0,0 +1,376 @@
+using fixed_width_column_wrapper = cudf::test::fixed_width_column_wrapper<T>;
+
+using strings_column_wrapper = cudf::test::strings_column_wrapper;
+
+using bool8 = cudf::experimental::bool8;
+
+// =============================================================================
+// ---- test data --------------------------------------------------------------
+
+namespace {
+namespace testdata {
+
+// ----- most numerics
+
+template<typename T>
+auto ascending() {
+    return fixed_width_column_wrapper<T>({ std::numeric_limits<T>::lowest(),
+                                            -100 -10, -1, 0, 1, 10, 100,
+                                            std::numeric_limits<T>::max() });
+}
+
+template<typename T>
+auto descending() {
+    return fixed_width_column_wrapper<T>({ std::numeric_limits<T>::max(),
+                                           100, 10, 1, 0, -1, -10, -100,
+                                           std::numeric_limits<T>::lowest() });
+}
+
+template<typename T>
+auto empty() {
+    return fixed_width_column_wrapper<T>({ });
+}
+
+template<typename T>
+auto nulls_after() {
+    return fixed_width_column_wrapper<T>({ 0, 0 }, { 1, 0 });
+}
+
+template<typename T>
+auto nulls_before() {
+    return fixed_width_column_wrapper<T>({ 0, 0 }, { 0, 1 });
+}
+
+// ----- bool8
+
+template<>
+auto ascending<bool8>() {
+    return fixed_width_column_wrapper<bool8>({ false, true });
+}
+
+template<>
+auto descending<bool8>() {
+    return fixed_width_column_wrapper<bool8>({ true, false });
+}
+
+// ----- std::string
+
+template<>
+auto ascending<std::string>() {
+    return strings_column_wrapper({ "A", "B" });
+}
+
+template<>
+auto descending<std::string>() {
+    return strings_column_wrapper({ "B", "A" });
+}
+
+template<>
+auto empty<std::string>() {
+    return strings_column_wrapper({ });
+}
+
+template<>
+auto nulls_after<std::string>() {
+    return strings_column_wrapper({ "identical", "identical" }, { 1, 0 });
+}
+
+template<>
+auto nulls_before<std::string>() {
+    return strings_column_wrapper({ "identical", "identical" }, { 0, 1 });
+}
+    
+} // namespace testdata
+} // anonymous namespace
+
+// =============================================================================
+// ---- tests ------------------------------------------------------------------
+
+template <typename T>
+struct IsSortedNumeric : public cudf::test::BaseFixture {};
+
+using test_types = ::testing::Types<int8_t, int16_t, int32_t, int64_t,
+                                    float, double,
+                                    cudf::experimental::bool8,
+                                    std::string>;
+
+// compiles and passes tests
+TYPED_TEST_CASE(IsSortedNumeric, cudf::test::NumericTypes);
+
+// compiles and passes tests
+// TYPED_TEST_CASE(IsSortedStrings, ::testing::Types<std::string>);
+
+// // compiles and passes tests if static_asserts for fixed-width types are removed
+// TYPED_TEST_CASE(IsSortedNumeric, test_types); 
```

Comment:
Instead of specializing for `std::string`, use `cudf::string_view` and use the `cudf::test::AllTypes`. Then it should work.

Reply:
`cudf::test::AllTypes` doesn't include `string_view`, but it does include `TimestampTypes`, which these tests do not support. I've updated to this which works flawlessly:
```
using test_types = Concat<NumericTypes, Types<cudf::string_view>>;
TYPED_TEST_CASE(IsSortedTest, test_types);
```

Reply:
> 1) add string_view to AllTypes

I think this makes sense.

> 2) make these tests work with timestamps 

Why don't they support timestamps? As far as sorting is concerned, I'd think they'd be no different from a numeric type.

Reply:
The legacy tests did not support time stamps, so I didn't add them (and they compiled but did not pass - now that I'm thinking about it, it might be due negative integers). Also, I didn't see any examples on how to use timestamps so far. At least not in tests. I'll take another shot at it.

Reply:
It may be worth adding `LexicographicalTypes`.

Reply:
Adding `string_view` to `AllTypes` breaks other tests (expected). I've added `LexicographicalTypes`, and I'm using that in the tests instead. `AllTypes` should have `string_view`, but other tests need updating for that to happen. Advice?

Reply:
https://github.com/rapidsai/cudf/pull/3299#discussion_r345354714